src/App.tsx

```diff
@@ -1,61 +1,115 @@
+import { ChangeEvent, FormEventHandler, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+let highestId = Math.max.apply(null, todosFromServer.map(todo => todo.id));
 
 export const App = () => {
+  const [newTaskTitle, setNewTaskTitle] = useState('');
+  const [newTaskUserId, setNewTaskUserId] = useState(0);
+  const [hasEmptyTitle, setHasEmptyTitle] = useState(false);
+  const [hasEmptyUser, setHasEmptyUser] = useState(false);
+
+  const todos: Todo[] = todosFromServer.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
+
+  const addNewTask: FormEventHandler = (event) => {
+    event.preventDefault();
+
+    setHasEmptyTitle(!newTaskTitle);
+    setHasEmptyUser(!newTaskUserId);
+
+    if (newTaskTitle && newTaskUserId) {
+      highestId += 1;
+      todosFromServer.push({
+        title: newTaskTitle,
+        completed: false,
+        userId: newTaskUserId,
+        id: highestId,
+      });
+      setNewTaskTitle('');
+      setNewTaskUserId(0);
+    }
+  };
+
+  const validateTitleField = (event: ChangeEvent<HTMLInputElement>) => {
+    const validatedTitle
```

Comment:
Use a more descriptive function name for `validateTitleField` such as `handleTitleChange`.